Core/src/com/serotonin/m2m2/Common.java

```diff
@@ -453,6 +453,14 @@ public static PermissionHolder getUser() throws PermissionException {
         throw new PermissionException(new TranslatableMessage("permission.exception.noAuthenticationSet"), null);
     }
 
+    public static PermissionHolder getPermissionHolder() {
+        PermissionHolder permissionHolder = null;
+        try {
+            permissionHolder = getUser();
+        } catch (PermissionException ignored) {}
+        return permissionHolder;
+    }
+
     public static Path getTempPath() {
```

Comment:
I would avoid adding this method, we should always have a PermissionHolder available. If we are trying to find out the current user and there isn't one, its an error and we want to know about it.

Reply:
@jazdw can you check my last 2 commit,

Looks like using strategy `MODE_THREADLOCAL` is removing authentication for the child threads.
If task is created from a child thread, it doesn't have the user available (Authentication is null).
I have fixed the tests by adding admin user but also had to change strategy to `MODE_INHERITABLETHREADLOCAL`

So tests that are using to create tasks from child thread or even JUnit timeout tests have no user available.

Tests are fixed but do we need change default strategy to `MODE_INHERITABLETHREADLOCAL`?
If we have some tasks in real application which created in child thread(s) will throw error as well.

Or should we ignore the user and use null Authentication for this child threads as before ?

Reply:
> Tests are fixed but do we need change default strategy to MODE_INHERITABLETHREADLOCAL?
I don't think we should be setting the mode to inheritable for the tests or for the real application.

The idea is that `MODE_THREADLOCAL` will not implicitly propagate the security context to new threads. You must explicitly set the security context in any new threads which are spawned. We need to figure out where these new threads are coming from and do something like `DelegatingSecurityContextExecutorService` or `Task` which explicitly set the security context in the new thread. Lets get on a call today and discuss.